nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -214,106 +276,178 @@ public void onScheduled(final ProcessContext context) {
+                    axisFault);
             throw new ProcessException(axisFault);
         }
     }
 
+    protected OMElement createAuthHeader(OMElement parent, Map<String, Object> map) {
+        OMElement element = null;
+        if (map != null) {
+            for (String key : map.keySet()) {
+                element = fac.createOMElement(key, omNamespace);
+                Object value = map.get(key);
+                if (value instanceof String) {
+                    element.setText(value.toString());
+                    if (parent != null) {
+                        parent.addChild(element);
+                    }
+                } else {
+                    OMElement childElement = createAuthHeader(element, (Map) value);
+                    element.addChild(childElement);
+                }
+            }
+        }
+        return element;
+    }
+
     @OnStopped
     public void onStopped(final ProcessContext context) {
         try {
             serviceClient.cleanup();
-        } catch (AxisFault axisFault) {
+        }
+        catch (AxisFault axisFault) {
             getLogger().error("Failed to clean up the web service client.", axisFault);
             throw new ProcessException(axisFault);
         }
     }
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
 
-        final StopWatch stopWatch = new StopWatch(true);
-        //get the dynamic properties, execute the call and return the results
-        OMFactory fac = OMAbstractFactory.getOMFactory();
-        OMNamespace omNamespace = fac.createOMNamespace(context.getProperty(WSDL_URL).getValue(), "nifi");
+        final String methodName = context.getProperty(METHOD_NAME)
+                                         .evaluateAttributeExpressions(flowFile)
+                                         .getValue();
+
+        // TODO: Not sure if this will apply in all cases. Some services require SOAPAction, some don't.
+        serviceClient.getOptions().setAction(apiNamespace + methodName);
 
-        final OMElement method = getSoapMethod(fac, omNamespace, context.getProperty(METHOD_NAME).getValue());
+        //get the dynamic properties, execute the call and return the results
+        final OMElement method = getSoapMethod(fac, omNamespace, methodName);
 
         //now we need to walk the arguments and add them
-        addArgumentsToMethod(context, fac, omNamespace, method);
+        addArgumentsToMethod(context, fac, omNamespace, method, flowFile);
         final OMElement result = executeSoapMethod(method);
-        final FlowFile flowFile = processSoapRequest(session, result);
+        getLogger().debug("RESULT" + result);
+        flowFile = processSoapRequest(session, result, flowFile);
+
+        // TODO: should add a failure situation?
```

Comment:
Convention for Ingress processors is to have a single relationship and then to rollback the the session if there is a failure. See the GetHTTP processor for an example